src/test/java/net/smartcosmos/dao/objects/impl/ObjectPersistenceServiceTest.java

```diff
@@ -108,8 +113,208 @@ public void getObjects() throws Exception {
+
+        queryParams.put(ObjectPersistenceService.TYPE, "type z");
+        response = objectPersistenceService
+            .findByQueryParameters(accountUrn, queryParams);
+        assertTrue(response.size() == 0);
+
+        queryParams.put(ObjectPersistenceService.TYPE, "ype");
+        response = objectPersistenceService
+            .findByQueryParameters(accountUrn, queryParams);
+        assertTrue(response.size() == 0);
+
+        queryParams.remove(ObjectPersistenceService.TYPE);
+        queryParams.put(ObjectPersistenceService.MONIKER_LIKE, "moniker");
+        response = objectPersistenceService
+            .findByQueryParameters(accountUrn, queryParams);
+        assertTrue(response.size() == 3);
+
+        queryParams.put(ObjectPersistenceService.MONIKER_LIKE, "moniker o");
+        response = objectPersistenceService
+            .findByQueryParameters(accountUrn, queryParams);
+        assertTrue(response.size() == 1);
+
+        queryParams.put(ObjectPersistenceService.MONIKER_LIKE, "moniker t");
+        response = objectPersistenceService
+            .findByQueryParameters(accountUrn, queryParams);
+        assertTrue(response.size() == 2);
+
+        queryParams.put(ObjectPersistenceService.MONIKER_LIKE, "moniker three");
+        response = objectPersistenceService
+            .findByQueryParameters(accountUrn, queryParams);
+        assertTrue(response.size() == 1);
+
+        queryParams.put(ObjectPersistenceService.MONIKER_LIKE, "moniker z");
+        response = objectPersistenceService
+            .findByQueryParameters(accountUrn, queryParams);
+        assertTrue(response.size() == 0);
+
+        queryParams.put(ObjectPersistenceService.MONIKER_LIKE, "oniker");
+        response = objectPersistenceService
+            .findByQueryParameters(accountUrn, queryParams);
+        assertTrue(response.size() == 0);
+
+    }
+
+    @Test
+    public void findByQueryParametersLastModified() throws Exception
+    {
+        final UUID accountUuid = UuidUtil.getNewUuid();
+        final String accountUrn = UuidUtil.getAccountUrnFromUuid(accountUuid);
+        Map<String, Object> queryParams = new HashMap<>();
+
+        Long firstDate = new Date().getTime();
+        Thread.sleep(DELAY_BETWEEN_LAST_MODIFIED_DATES);
+
+        ObjectEntity firstObject = ObjectEntity.builder().accountId(accountUuid)
+            .objectUrn("objectUrnLastModTest1").name("last mod test 1").type("anythingIsFine").build();
+        this.objectRepository.save(firstObject);
+
+        Long secondDate = new Date().getTime();
+        Thread.sleep(DELAY_BETWEEN_LAST_MODIFIED_DATES);
+
+        ObjectEntity secondObject = ObjectEntity.builder().accountId(accountUuid)
+            .objectUrn("objectUrnLastModTest2").name("last mod test 2").type("anythingIsFine").build();
+        this.objectRepository.save(secondObject);
+
+        Long thirdDate = new Date().getTime();
+        Thread.sleep(DELAY_BETWEEN_LAST_MODIFIED_DATES);
```

Comment:
I'm not thinking of it - anything else would involve doing something evil to directly hack lastModifiedTimestamp. On the bright side, I've cut the delays down to 10 ms, so the additional cost is now 30 ms. Still working fine. Canned data seems to me a) more complicated and b) in violation of the unit test's independence as a piece of code.